src/librustc_ast_lowering/item.rs

```diff
@@ -719,9 +719,9 @@ impl<'hir> LoweringContext<'_, 'hir> {
             VariantData::Tuple(ref fields, id) => hir::VariantData::Tuple(
                 self.arena
                     .alloc_from_iter(fields.iter().enumerate().map(|f| self.lower_struct_field(f))),
-                self.lower_node_id(id, DUMMY_SP),
+                self.lower_node_id(id, span),
             ),
-            VariantData::Unit(id) => hir::VariantData::Unit(self.lower_node_id(id, DUMMY_SP)),
+            VariantData::Unit(id) => hir::VariantData::Unit(self.lower_node_id(id, span)),
```

Comment:
These fixes on lines 722 and 724 seem important, or at least worth considering trying to land on their own in a separate PR?

That is, it seems like *this* could cause an interesting change to the diagnostic output in some cases, and it would be good to vet that change independently of the others changes in this PR, which I had hoped would be a refactoring....

(I understand the temptation to fold fixes like this in as a drive-by change, especially since I'm not sure how best to make this change without making the other changes to e.g. thread the `span` argument around. But I'm worried about coupling it with such a big change overall, especially one that we might choose to revert later on as we gather more experience with the change...)

Reply:
The behaviour does not change here. This choice of span for the VariantData is the one already implemented in `hir::map::NodeCollector::visit_variant`.